dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/enrollment/EnrollmentService.java

```diff
@@ -58,6 +59,15 @@ List<Enrollment> getEnrollments(EnrollmentOperationParams params)
   Page<Enrollment> getEnrollments(EnrollmentOperationParams params, PageParams pageParams)
       throws BadRequestException, ForbiddenException;
 
+  /**
+   * Returns a list of existing Enrollments from the provided UIDs
+   *
+   * @param uids Enrollment UIDs to check
+   * @return Enrollment list
+   */
+  List<Enrollment> getEnrollments(@Nonnull List<String> uids, UserDetails currentUser)
```

Comment:
@muilpp and @enricocolasante will this not be called from an HTTP request? If so the current user should be set in the auth context and we do not need to pass it explicitly.

Reply:
shouldn't we always get it from the context, even in those cases where it's provided in the request?

Reply:
That is something we have to discuss and settle on. Right now with SMS its not in the auth context so you cannot get it that way. Deduplication AFAIK is only triggered via HTTP, hence you should not have to pass the user.

I would also hope that you can reuse the existing service method to get some enrollments.

Reply:
With the exception of SMS, I think now it makes more sense to always get it from the context, before there was a call to the database, but now if we use `UserDetails` there's none anymore.

Reply:
We need to find our way to proceed but if we know how the method is used, I would not pass the user as a parameter and get it from the context.